src/audio/google/google_ctc_audio_processing.c

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Google LLC.
+//
+// Author: Eddy Hsu <eddyhsu@google.com>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/data_blob.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <rtos/init.h>
+
+#include <google_ctc_audio_processing.h>
+#include <google_ctc_audio_processing_sof_message_reader.h>
+
+#include "google_ctc_audio_processing.h"
+
+LOG_MODULE_REGISTER(google_ctc_audio_processing, CONFIG_SOF_LOG_LEVEL);
+
+/* bf0e1bbc-dc6a-45fe-bc90-2554cb137ab4 */
+DECLARE_SOF_RT_UUID("google-ctc-audio-processing", google_ctc_audio_processing_uuid,
+		    0xbf0e1bbc, 0xdc6a, 0x45fe, 0xbc, 0x90, 0x25, 0x54, 0xcb,
+		    0x13, 0x7a, 0xb4);
```

Comment:
This could now be moved to the new UUID system @andyross added - see commit 17f46a71ee22505949af5d2de8def513b16b7021 and 51a5fb9c8098207527a4ef8b5afbc78a6f3b9bda